lib/cocoapods/installer/xcode/target_validator.rb

```diff
@@ -108,6 +109,31 @@ def verify_no_pods_used_with_multiple_swift_versions
               "that do not have the same Swift version:\n\n#{error_messages.join("\n")}"
           end
         end
+
+        def verify_swift_pods_have_module_dependencies
+          error_messages = []
+          pod_targets.each do |pod_target|
+            next unless pod_target.uses_swift?
+
+            non_module_dependencies = []
+            pod_target.dependent_targets.each do |dependent_target|
+              next if !dependent_target.should_build? || dependent_target.defines_module?
+              non_module_dependencies << dependent_target.name
+            end
+
+            next if non_module_dependencies.empty?
+
+            error_messages << "The swift pod `#{pod_target.name}` depends upon #{non_module_dependencies.to_sentence}, " \
+                              'which do not define modules. ' \
+                              'To opt into those targets generating module maps '\
+                              '(which is necessary to import them from swift when building as static libraries), ' \
+                              "you may add `'DEFINES_MODULE' => 'YES'` to those pods' `pod_target_xcconfig`."
```

Comment:
I haven't tried this yet, however - a couple questions:

1. What if you don't control the pod in question
2. What about the module map file, in this case?

Reply:
> What if you don't control the pod in question

It needs to be updated to opt-in, for now. We can explore making this an option in the Podfile down the road, but as that's purely new funcitonality I dont consider it a blocker

> What about the module map file, in this case?
 
The module map file will only be generated if:

1) The pod is consumed as a framework
2) The pod contains swift
3) `'DEFINES_MODULE' => 'YES'`

This is because adding a modulemap for a pod that has not been updated has the potential to break imports of that pods headers